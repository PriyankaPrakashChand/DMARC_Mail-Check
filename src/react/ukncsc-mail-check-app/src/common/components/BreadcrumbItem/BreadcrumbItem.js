import React from 'react';
import { Breadcrumb } from 'semantic-ui-react';
import { Link } from 'react-router-dom';

const BreadcrumbItem = ({ link, children, ...props }) => (
  <React.Fragment>
    <Breadcrumb.Divider icon="right angle" />
    <Breadcrumb.Section {...props}>
      {link ? <Link to={link}>{children}</Link> : children}
    </Breadcrumb.Section>
  </React.Fragment>
);

export default BreadcrumbItem;
