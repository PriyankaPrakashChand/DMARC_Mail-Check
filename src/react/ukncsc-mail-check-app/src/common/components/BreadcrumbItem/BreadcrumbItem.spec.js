import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from 'react-testing-library';
import BreadcrumbItem from './BreadcrumbItem';

describe('BreadcrumbItem', () => {
  let container;

  test('it should match the snapshot', () => {
    ({ container } = render(
      <MemoryRouter>
        <BreadcrumbItem>ncsc.gov.uk</BreadcrumbItem>
      </MemoryRouter>
    ));

    expect(container).toMatchSnapshot();
  });
});
