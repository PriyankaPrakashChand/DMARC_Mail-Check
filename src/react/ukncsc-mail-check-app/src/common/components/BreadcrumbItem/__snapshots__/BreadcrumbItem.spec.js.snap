// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbItem it should match the snapshot 1`] = `
<div>
  <i
    aria-hidden="true"
    class="right angle icon divider"
  />
  <div
    class="section"
  >
    ncsc.gov.uk
  </div>
</div>
`;
